crates/bevy_ecs/src/world/component_constants.rs

```diff
@@ -13,6 +13,8 @@ pub const ON_INSERT: ComponentId = ComponentId::new(1);
 pub const ON_REPLACE: ComponentId = ComponentId::new(2);
 /// [`ComponentId`] for [`OnRemove`]
 pub const ON_REMOVE: ComponentId = ComponentId::new(3);
+/// [`ComponentId`] for [`OnDespawn`]
+pub const ON_DESPAWN: ComponentId = ComponentId::new(4);
```

Comment:
This doesn't seem to have been added to `World::bootstrap`, so the component ID won't be set correctly.

Reply:
Hmm interesting that the relevant recursive despawn tests still pass. Probably just getting lucky on the id allocation :)

I'll fix this.